documentation/deprecated_features.markdown

```diff
@@ -0,0 +1,349 @@
+`authorization-required` is set to `false` or `client-whitelist` has one or
+more entries, these settings would be used to authorize access to the
+`puppet-admin` HTTP API instead of `trapperkeeper-authorization`.
+
+If `authorization-required` is set to `true` or is not specified and if
+the `client-whitelist` was empty, you could just remove the `puppet-admin`
+section from your configuration and restart your puppetserver service in order
+for Puppet Server to start using the `trapperkeeper-authorization` "auth.conf"
+file.  The only behavior that would change in Puppet Server from doing this
+would be that a warning message would no longer be written to the
+puppetserver.log file.
+
+If `authorization-required` is set to `false`, you would need to create
+corresponding rules in the `trapperkeeper-authorization` file which would
+allow unauthenticated client access to the "puppet-admin" API endpoints.
+
+For example:
+
+~~~~hocon
+authorization: {
+    version: 1
+    rules: [
+            {
+                match-request: {
+                    path: "/puppet-admin-api/v1/environment-cache"
+                    type: path
+                    method: delete
+                }
+                allow-unauthenticated: true
+                sort-order: 200
+                name: "environment-cache"
+            },
+            {
+                match-request: {
+                    path: "/puppet-admin-api/v1/jruby-pool"
+                    type: path
+                    method: delete
+                }
+                allow-unauthenticated: true
+                sort-order: 200
+                name: "jruby-pool"
+            },
+            ...
+     ]
+}
+~~~~
+
+If `authorization-required` is set to `true` or not set but the
+`client-whitelist` has one or more custom entries in it, you would need to
+create corresponding rules in the `trapperkeeper-authorization` "auth.conf"
+file which would allow only specific clients access to the "puppet-admin"
+API endpoints.
+
+For example, the current "puppet-admin" configuration could have:
+
+~~~~hocon
+puppet-admin: {
+    client-whitelist: [ admin1, admin2 ]
+}
+~~~~
+
+Corresponding `trapperkeeper-authorization` rules could have:
+
+~~~~hocon
+authorization: {
+    version: 1
+    rules: [
+            {
+                match-request: {
+                    path: "/puppet-admin-api/v1/environment-cache"
+                    type: path
+                    method: delete
+                }
+                allow: [ admin1, admin2 ]
+                sort-order: 200
+                name: "environment-cache"
+            },
+            {
+                match-request: {
+                    path: "/puppet-admin-api/v1/jruby-pool"
+                    type: path
+                    method: delete
+                }
+                allow: [ admin1, admin2 ]
+                sort-order: 200
+                name: "jruby-pool"
+                },
```

Comment:
@nwolfe Thanks, I fixed this up, just squashing it into the prior commit - now as 912cb21 - since it was cosmetic.